app/code/community/Oyst/OneClick/Model/OneClick/ApiWrapper.php

```diff
@@ -75,17 +79,22 @@ public function authorizeOrder($dataFormated)
             $notifications->addEvent('order.stock.released');
         }
 
-        $orderParams = $this->getOneClickOrderParams();
+        $orderParams = $this->getOneClickOrderParams($dataFormated);
+
+        $this->quoteId = Mage::getSingleton('checkout/session')->getQuote()->getId();
```

Comment:
This as to be above `$orderParams = $this->getOneClickOrderParams($dataFormated);` because in `getOneClickOrderParams()` you use `$this->quoteId`